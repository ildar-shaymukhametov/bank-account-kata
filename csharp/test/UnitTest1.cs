using ApprovalTests;
using ApprovalTests.Reporters;
using Xunit;

namespace test
{
    [UseReporter(typeof(DiffReporter))]
    public class UnitTest1
    {
        [Fact]
        public void Test1()
        {
            var sut = new Account(new FakeTimeProvider());
            sut.Deposit(1000);
            sut.Deposit(2000);
            sut.Withdraw(500);
            var output = sut.Print();
            Approvals.Verify(output);
        }
    }

    class FakeTimeProvider : ITimeProvider
    {
        private int count;

        public FakeTimeProvider()
        {
            this.count = 1;
        }

        public DateTime Now()
        {
            switch (this.count++)
            {
                case 1: return new DateTime(2012, 1, 10);
                case 2: return new DateTime(2012, 1, 13);
                case 3: return new DateTime(2012, 1, 14);
                default: throw new IndexOutOfRangeException();
            }
        }
    }
}