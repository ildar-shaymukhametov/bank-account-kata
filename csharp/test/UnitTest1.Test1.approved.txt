Date       || Amount || Balance
14/01/2012 || -500   || 2500
13/01/2012 || 2000   || 3000
10/01/2012 || 1000   || 1000